src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinition.java

```diff
@@ -127,15 +134,38 @@ public boolean isLightweight() {
         delegate.setPoll(true);
         delegate.setChangelog(true);
         try (WorkspaceList.Lease lease = computer.getWorkspaceList().acquire(dir)) {
-            delegate.checkout(build, dir, listener, node.createLauncher(listener));
-            FilePath scriptFile = dir.child(scriptPath);
-            if (!scriptFile.absolutize().getRemote().replace('\\', '/').startsWith(dir.absolutize().getRemote().replace('\\', '/') + '/')) { // TODO JENKINS-26838
-                throw new IOException(scriptFile + " is not inside " + dir);
-            }
-            if (!scriptFile.exists()) {
-                throw new AbortException(scriptFile + " not found");
+            for (int retryCount = getScmCheckoutRetryCount(); retryCount >= 0; retryCount--) {
+                try {
+                    delegate.checkout(build, dir, listener, node.createLauncher(listener));
+
+                    FilePath scriptFile = dir.child(scriptPath);
+                    if (!scriptFile.absolutize().getRemote().replace('\\', '/').startsWith(dir.absolutize().getRemote().replace('\\', '/') + '/')) { // TODO JENKINS-26838
+                        throw new IOException(scriptFile + " is not inside " + dir);
+                    }
+                    if (!scriptFile.exists()) {
+                        throw new AbortException(scriptFile + " not found");
+                    }
+                    script = scriptFile.readToString();
+                    break;
+                } catch (AbortException e) {
+                    // abort exception might have a null message.
+                    // If so, just skip echoing it.
+                    if (e.getMessage() != null) {
+                        listener.error(e.getMessage());
+                    }
+                } catch (InterruptedIOException e) {
+                    throw (InterruptedException)new InterruptedException().initCause(e);
+                } catch (IOException e) {
+                    // checkout error not yet reported
+                    listener.error(e.toString());
+                }
+                
+                if (retryCount == 0)   // all attempts failed
+                    throw new RunnerAbortedException();
```

Comment:
Use `AbortException` not `RunnerAbortedException`. Can probably rearrange the logic a bit so that if we are on the last iteration (the normal case), the original `AbortException` or whatever is rethrown unmodified.